sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2374,14 +2374,14 @@ object functions {
    * @group datetime_funcs
    * @since 1.5.0
    */
-  def date_add(start: Column, days: Int): Column = withExpr { DateAdd(start.expr, Literal(days)) }
+  def date_add(start: Column, days: Int): Column = withExpr { AddDays(start.expr, Literal(days)) }
```

Comment:
It's confusing to users that `date_add` add days to the given date and `add_months` add months to the given date. I think `add_days` and `add_months` are more consistent.

Other databases(e.g. MySQL, Postgres) only have `date_add` which adds interval to the given date, so that they don't need `add_days` and `add_months` respectively.

The function name is already realsed and maybe hard to change, but changing the expression name to match the really logic seems good.

@rxin any thoughts?